### PR TITLE
Attendance updates

### LINKED ISSUE #491: The number of students who submitted a code and past codes should be shown on the large group page
The delete feature should also be implemented

### PR DESCRIPTION
**Pull request by kmcnellis**
Split admin into settings, users, and attendance 

Add a list of past attendance codes to admin, along with deleting codes and number of attendees.

Split code generation and showing past codes into a common directive for small and large group.

Smallgroups can create bonus daycodes

Students can no longer query the classyear or small group and receive a list of all attendance codes (just dates)

fixes #491

### CODE REVIEW

File: client/components/viewAttendance/viewAttendance.directive.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          notify('Successfully removed day: ' + dateString);
+          $scope.group = group;
+          updateGroup();
+        })
+        .error(function() {
+          notify('ERROR: Could not remove day: ' + dateString);
+        });
+      };
+
+
```
Comment (agundy) [inline client/components/viewAttendance/viewAttendance.directive.js]: why all these spaces???

File: server/api/classyear/classyear.controller.js [hunk 2]
```diff
@@ -134,20 +158,39 @@ exports.daycode = function(req, res){
...
     });
   });
 };
 
+// Delete a day code from a classyear and the corresponding daycode submission from attendance
+// Restricted to admins
+// router.delete('/day/:dayCode', auth.hasRole('admin'), controller.deleteDay);
+exports.deleteDay = function(req, res){
+    var dayCode = req.params.dayCode;
+    var smallGroupId = req.params.id;
```
Comment (agundy) [inline server/api/classyear/classyear.controller.js]: extra line

File: server/api/user/index.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ router.get('/stats',  controller.publicStats);
 router.get('/past', controller.past);
 router.get('/search', controller.search);
 router.get('/adminstats', auth.hasRole('admin'), controller.allStats);
+router.get('/smallgroup', auth.isAuthenticated(), controller.smallgroup);
```
Comment (agundy) [inline server/api/user/index.js]: Two calls to controller.smallgroup one with params one without

**Comment by agundy** [comment]
small fixes! looks good then!

**Comment by agundy** [comment]
- after removing the attendance daycode it doesn't reshow the generate daycode or generate bonus daycode
- Show full screen doesn't hide everything and still shows a useless button

File: server/api/user/user.controller.js [hunk 4]
```diff
@@ -287,7 +287,7 @@ exports.privateProfile = function (req, res, next) {
  * Get a single user's smallgroup
  */
 exports.smallgroup = function (req, res, next) {
-  var userId = req.params.id;
+  var userId = req.user.id;
```
Comment (agundy) [inline server/api/user/user.controller.js]: check all endpoints calling this. Got an error loading the user profile

**Comment by kmcnellis** [comment]
Think I got everything

**Pull request merged**